sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,252 @@
+use aws_sdk_timestreamquery as timestream_query;
+use aws_sdk_timestreamquery::types;
+use aws_types::region::Region;
+use clap::Parser;
+use std::fs;
+use std::io::{self, Write};
+
+static DEFAULT_DATABASE_NAME: &str = "devops_multi_sample_application";
+static DEFAULT_OUTPUT_FILE: &str = "query_results.log";
+static DEFAULT_REGION: &str = "us-east-1";
+static DEFAULT_TABLE_NAME: &str = "host_metrics_sample_application";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Args {
+    // The Timestream for LiveAnalytics database name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_DATABASE_NAME)]
+    pub database_name: String,
+
+    // The full name of the output file to write query results to.
+    // For example, query_results.log
+    #[arg(short, long, default_value = DEFAULT_OUTPUT_FILE)]
+    pub output_file: String,
+
+    // The name of the AWS region to use for queries
+    #[arg(short, long, default_value = DEFAULT_REGION)]
+    pub region: String,
+
+    // The Timestream for LiveAnalytics table name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_TABLE_NAME)]
+    pub table_name: String,
+}
+
+pub async fn get_connection(
+    region: &String,
+) -> Result<timestream_query::Client, timestream_query::Error> {
+    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+        .region(Region::new(region.to_owned()))
+        .load()
+        .await;
+    let (client, reload) = timestream_query::Client::new(&config)
+        .with_endpoint_discovery_enabled()
+        .await
+        .expect("Failure");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> io::Result<()> {
+    let s_formatted = format!("{}\n", s);
+    file.write(s_formatted.as_bytes())?;
+    file.flush()?;
+    Ok(())
+}
+
+pub fn process_scalar_type(data: &types::Datum) -> Result<String, String> {
```

Comment:
We are returning a string on success or a string on error. I think we should handle the error appropriately and fail.

Reply:
By failing do you mean `panic!`? I'm assuming handling would be logging the error.

Reply:
Should all functions be changed to fail instead if an error occurs and not return a `Result`?

Reply:
`query_common` functions now all return a `Result` and will use `Box<dyn Error>` as the `Err` value.